packages/core/src/common/connection-status.ts

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable, optional } from 'inversify';
+import { IdleService } from './idle';
+import { ILogger } from '../common/logger';
+import { Event, Emitter } from '../common/event';
+
+/**
+ * Service for listening on backend connection changes.
+ */
+export const ConnectionStatusService = Symbol('ConnectionStatusService');
+export interface ConnectionStatusService {
```

Comment:
Do we need it in common? Could it be used in node?

Reply:
It should be easier to test without the browser.

Reply:
It will be testable without browser if it belongs to browser module as well?

Something should belong to common if it is supposed to be used in node and browser. If it is supposed to be used only in browser and electon-broswer, it should go to browser.

Reply:
> Something should belong to common if it is supposed to be used in node and browser

Exactly. It is an abstract service, you can use it from node too. If you implement the abstract methods.

Reply:
Do you think it is going to happen?

Reply:
Not at this point, but unless one can clearly state it will not happen, it can happen. It was easier for me to write tests without the browser. Did not check them in yet. Bear with me. I need to adjust two more things on the API.

Reply:
ok, just we do oppositive in other places, like planning moving commands, keybindings and menus to browser to expose apis from modules to which they belong